dexter/src/main/java/com/karumi/dexter/DexterInstance.java

```diff
@@ -52,18 +58,27 @@
    * @param listener The class that will be reported when the state of the permission is ready
    */
   void checkPermission(String permission, PermissionListener listener) {
-    if (isRequestingPermission.getAndSet(true)) {
-      throw new IllegalStateException(
-          "Only one permission request at a time. Currently handling permission: ["
-              + this.permission + "]");
-    }
+    MultiplePermissionsListener adapter =
+        new MultiplePermissionsListenerToPermissionListenerAdapter(listener);
+    checkPermissions(Collections.singleton(permission), adapter);
+  }
+
+  /**
+   * Checks the state of a collection of permissions reporting their state to the listener when all
+   * of them are resolved
+   *
+   * @param permissions Collection of values found in {@link android.Manifest.permission}
+   * @param listener The class that will be reported when the state of all the permissions is ready
+   */
+  void checkPermissions(Collection<String> permissions, MultiplePermissionsListener listener) {
+    assertNoDexterRequestOngoing();
+    assertRequestSomePermission(permissions);
```

Comment:
These methods are not using java ``assert`` methods in the implementation, maybe we should change these method names to something more related to ``validate...`` or ``check..``. I think ``checkNotAlreadyRequestingPermissions()`` or some similar name.